OpenSim/Actuators/Test/testActuators.cpp

```diff
@@ -522,6 +449,82 @@ void testClutchedPathSpring()
 }
 
 
+void testMcKibbenActuator()
+{
+
+    double pressure = 5 * 10e5; // 5 bars
+    double num_turns = 1.5;     // 1.5 turns
+    double B = 277.1 * 10e-4;  // 277.1 mm
+
+    using namespace SimTK;
+    std::clock_t startTime = std::clock();
+
+    double mass = 1;
+    double ball_radius = 10e-6;
+
+    Model *model = new Model;
+    model->setGravity(Vec3(0));
+
+    Ground& ground = model->updGround();
+
+    McKibbenActuator *actuator = new McKibbenActuator("mckibben", num_turns, B);
+
+    OpenSim::Body* ball = new OpenSim::Body("ball", mass, Vec3(0), mass*SimTK::Inertia::sphere(0.1));
+    ball->scale(Vec3(ball_radius), false);
+
+    actuator->addNewPathPoint("mck_ground", ground, Vec3(0));
+    actuator->addNewPathPoint("mck_ball", *ball, Vec3(ball_radius));
+
+    Vec3 locationInParent(0, ball_radius, 0), orientationInParent(0), locationInBody(0), orientationInBody(0);
+    SliderJoint *ballToGround = new SliderJoint("ballToGround", ground, locationInParent, orientationInParent, *ball, locationInBody, orientationInBody);
+
+    ballToGround->updCoordinate().setName("ball_d");
+    ballToGround->updCoordinate().setPrescribedFunction(LinearFunction(20 * 10e-4, 0.5 * 264.1 * 10e-4));
+    ballToGround->updCoordinate().set_prescribed(true);
+
+    model->addBody(ball);
+    model->addJoint(ballToGround);
+    model->addForce(actuator);
+
+    PrescribedController* controller = new PrescribedController();
+    controller->addActuator(*actuator);
+    controller->prescribeControlForActuator("mckibben", new Constant(pressure));
+
+    model->addController(controller);
+
+    ForceReporter* reporter = new ForceReporter(model);
+    model->addAnalysis(reporter);
+
+    SimTK::State& si = model->initSystem();
+
+    model->getMultibodySystem().realize(si, Stage::Position);
+
+    double final_t = 10.0;
+    double nsteps = 10;
+    double dt = final_t / nsteps;
+
+    Manager manager(*model);
+    manager.setIntegratorAccuracy(1e-7);
+    si.setTime(0.0);
+    manager.initialize(si);
+
+    for (int i = 1; i <= nsteps; i++) {
+        si = manager.integrate(dt*i);
+        model->getMultibodySystem().realize(si, Stage::Velocity);
+        Vec3 pos = ball->findStationLocationInGround(si, Vec3(0));
+
+        double applied = actuator->computeActuation(si);;
+
+        double theoretical = (pressure / (4 * pow(num_turns, 2) * SimTK::Pi)) * (3 * pow(pos(0), 2) - pow(B, 2));
+
+        ASSERT_EQUAL(applied, theoretical, 10.0);
+    }
+
+
+    std::cout << " ******** Test McKibbenActuator time = ********" <<
+        1.e3*(std::clock() - startTime) / CLOCKS_PER_SEC << "ms\n" << endl;
+}
+
```

Comment:
github is not showing this side by side, was there material change other than moving text around?

Reply:
no change.